pkg/autoscaler/autoscaler/tidb_autoscaler.go

```diff
@@ -43,16 +43,16 @@ func (am *autoScalerManager) syncTiDB(tc *v1alpha1.TidbCluster, tac *v1alpha1.Ti
 		return nil
 	}
 	var targetReplicas int32
-	if tac.Spec.TiDB.ExternalEndpoint.Endpoint == nil {
-		instances := filterTidbInstances(tc)
-		targetReplicas, err = calculateTidbMetrics(tac, sts, instances)
+	if tac.Spec.TiDB.ExternalEndpoint != nil && tac.Spec.TiDB.ExternalEndpoint.Endpoint != nil {
```

Comment:
Please help change https://github.com/pingcap/tidb-operator/blob/master/pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go#L177 to non-pointer.
Sorry that I did not consider it carefully enough during the design.